File: server/app/models/recommendation.py
from datetime import datetime
from typing import Optional

from .camel_model import CamelModel
from .song import Song


class Recommendation(Song):
    votes: list[str] = []


class RecommendationList(CamelModel):
    recommendations: list[Recommendation] = []
    voting_start_time: Optional[datetime] = None

File: client/src/types/Recommendation.ts
import { Song } from './Song';

export class Recommendation extends Song {
    votes: string[];

    constructor(data: Recommendation) {
        super(data);
        this.votes = data.votes;
    }
}

export class RecommendationList {
    recommendations: Recommendation[];
    votingStartTime: Date;

    constructor(data: RecommendationList) {
        this.recommendations = data.recommendations;
        this.votingStartTime = data.votingStartTime
    }
}